ml-prims/test/dist_adj.cu

```diff
@@ -78,25 +73,27 @@ public:
     allocate(y, n * k);
     allocate(dist_ref, m * n);
     allocate(dist, m * n);
-    r.uniform(x, m * k, T(-1.0), T(1.0));
-    r.uniform(y, n * k, T(-1.0), T(1.0));
-    InParams<T> in_params = {params.eps};
-    naiveDistanceAdj(dist_ref, x, y, m, n, k, params.eps);
+    r.uniform(x, m * k, DataType(-1.0), DataType(1.0));
+    r.uniform(y, n * k, DataType(-1.0), DataType(1.0));
+
+    DataType threshold = params.eps;
+
+    naiveDistanceAdj(dist_ref, x, y, m, n, k, threshold);
     char *workspace = nullptr;
-    size_t worksize = 0;
-    typedef cutlass::Shape<8, 128, 128> OutputTile_t;
-    distance<T, T, bool, InParams<T>, OutputTile_t>(
-      x, y, dist, m, n, k, in_params, EucExpandedL2, nullptr, worksize);
+    size_t worksize = getWorkspaceSize<EucExpandedL2, DataType, DataType, bool>(x, y, m, n, k);
     if (worksize != 0) {
       allocate(workspace, worksize);
     }
-    auto fin_op = [] __device__(T d_val, int g_d_idx,
-                                const InParams<T> &in_params) {
-      return d_val <= in_params.threshold;
+
+    typedef cutlass::Shape<8, 128, 128> OutputTile_t;
+
+    auto fin_op = [threshold] __device__(DataType d_val, int g_d_idx) {
+      (d_val <= threshold) ? (d_val = 1.f) : (d_val = 0.f);
```

Comment:
knitpick! maybe better to simplify this whole lambda itself as: `return d_val <= threshold;`?